applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/eigenfrequency_response_function_utility.h

```diff
@@ -314,33 +315,42 @@ class EigenfrequencyResponseFunctionUtility
                         gradient_contribution[coord_dir_i] += gradient_prefactors[i] * inner_prod(eigenvectors_of_element[i] , aux_vector) * mWeightingFactors[i];
                     }
 
-                   node_i.GetInitialPosition()[coord_dir_i] -= mDelta;
+                    node_i.GetInitialPosition()[coord_dir_i] -= mDelta;
+                    node_i.Coordinates()[coord_dir_i] -= mDelta;
                 }
                 noalias(node_i.FastGetSolutionStepValue(SHAPE_SENSITIVITY)) += gradient_contribution;
             }
         }
     }
 
     // --------------------------------------------------------------------------
-    double GetEigenvalue(int eigenfrequency_id)
+    double GetEigenvalue(const int& eigenfrequency_id)
     {
         return (mrModelPart.GetProcessInfo()[EIGENVALUE_VECTOR])[eigenfrequency_id-1];
     }
 
     // --------------------------------------------------------------------------
-    void DetermineEigenvectorOfElement(ModelPart::ElementType& traced_element, int eigenfrequency_id, int size_of_eigenvector, Vector& rEigenvectorOfElement)
+    void DetermineEigenvectorOfElement(ModelPart::ElementType& rElement, const int& eigenfrequency_id, Vector& rEigenvectorOfElement, ProcessInfo& CurrentProcessInfo)
     {
-        rEigenvectorOfElement.resize(size_of_eigenvector,false);
+        std::vector<std::size_t> eq_ids;
+        rElement.EquationIdVector(eq_ids, CurrentProcessInfo);
 
-        const std::size_t num_nodes = traced_element.GetGeometry().size();
-        const std::size_t num_node_dofs = size_of_eigenvector/num_nodes;
+        if (rEigenvectorOfElement.size() != eq_ids.size())
+        {
+            rEigenvectorOfElement.resize(eq_ids.size(), false);
+        }
 
-        for (std::size_t node_index=0; node_index<num_nodes; node_index++)
+        for (auto& r_node_i : rElement.GetGeometry())
```

Comment:
const?

Reply:
No, the `GetDofs()` function does not have the const version.